### PR TITLE
fix(GuildEditData): some fields can be null

### LINKED ISSUE #7631: Typings
### Which package is this bug report for?

discord.js

### Issue description

Removing afkChannel, systemChannel, rulesChannel or publicUpdatesChannel is made harder in typescript because of poorly written type definitions.
Also applies to channel's parent property.

### Code sample

```typescript
// TS2322: Type 'null' is not assignable to type 'VoiceChannelResolvable | undefined'.
await guild.edit({ afkChannel: null });

// translates to guild.edit({ afkChannel: null })
await guild.setAfkChannel(null);

// this works but in certain use cases forces multiple edit calls 
// if you're trying to edit other properties at once
```

### Package version

discord.js@13.6.0

### Node.js version

v16.9.0

### Operating system

_No response_

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

Guilds

### I have tested this issue on a development release

_No response_

### PR DESCRIPTION
**Pull request by almeidx**
**Please describe the changes this PR makes and why it should be merged:**
Closes #7631

https://github.com/discordjs/discord.js/blob/dcd479767b6ec980a373f2ea1f22754f41661c1e/packages/discord.js/typings/index.d.ts#L1013
https://github.com/discordjs/discord.js/blob/dcd479767b6ec980a373f2ea1f22754f41661c1e/packages/discord.js/typings/index.d.ts#L1032
https://github.com/discordjs/discord.js/blob/dcd479767b6ec980a373f2ea1f22754f41661c1e/packages/discord.js/typings/index.d.ts#L1030
https://github.com/discordjs/discord.js/blob/dcd479767b6ec980a373f2ea1f22754f41661c1e/packages/discord.js/typings/index.d.ts#L1029

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/discord.js/src/structures/Guild.js [hunk 1]
```diff
@@ -748,11 +748,11 @@ class Guild extends AnonymousGuild {
    * @property {DefaultMessageNotificationLevel|number} [defaultMessageNotifications] The default message notification
    * level of the guild
    * @property {SystemChannelFlagsResolvable} [systemChannelFlags] The system channel flags of the guild
-   * @property {TextChannelResolvable} [rulesChannel] The rules channel of the guild
-   * @property {TextChannelResolvable} [publicUpdatesChannel] The community updates channel of the guild
+   * @property {?TextChannelResolvable} [rulesChannel] The rules channel of the guild
+   * @property {?TextChannelResolvable} [publicUpdatesChannel] The community updates channel of the guild
    * @property {string} [preferredLocale] The preferred locale of the guild
```
Comment (Jiralite) [inline packages/discord.js/src/structures/Guild.js:753]: This is also nullable. It resets the locale back to English (US).

File: packages/discord.js/src/structures/Guild.js [hunk 2]
```diff
@@ -737,8 +737,8 @@ class Guild extends AnonymousGuild {
    * @property {string} [name] The name of the guild
    * @property {VerificationLevel|number} [verificationLevel] The verification level of the guild
    * @property {ExplicitContentFilterLevel|number} [explicitContentFilter] The level of the explicit content filter
```
Comment (Jiralite) [inline packages/discord.js/src/structures/Guild.js:739]: This is also nullable. It resets the explicit media content filter back to not scanning anything (0).

File: packages/discord.js/src/structures/Guild.js [hunk 3]
```diff
@@ -737,8 +737,8 @@ class Guild extends AnonymousGuild {
    * @property {string} [name] The name of the guild
    * @property {VerificationLevel|number} [verificationLevel] The verification level of the guild
```
Comment (Jiralite) [inline packages/discord.js/src/structures/Guild.js:738]: This is also nullable. It resets the verification level back to none (0).

File: packages/discord.js/src/structures/Guild.js [hunk 4]
```diff
@@ -748,11 +748,11 @@ class Guild extends AnonymousGuild {
    * @property {DefaultMessageNotificationLevel|number} [defaultMessageNotifications] The default message notification
```
Comment (Jiralite) [inline packages/discord.js/src/structures/Guild.js:748]: This is also nullable. It resets the default message notification level back to all messages (0).

**Review by Jiralite — COMMENTED** [review]
There are still more fields that are nullable that I have found from [here](https://discord.com/developers/docs/resources/guild#modify-guild). I have tested & verified them.

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**